src/main/java/gov/usgs/util/CryptoUtils.java

```diff
@@ -76,60 +79,37 @@
 	/** Number of bits for RSA 4096 bit key. */
 	public static final int RSA_4096 = 4096;
 
-	/**
-	 * Generate a signature.
-	 *
-	 * @param algorithm
-	 *            signature algorithm to use, as accepted by
-	 *            Signature.getInstance.
-	 * @param privateKey
-	 *            private key to use, should be acceptable by signature
-	 *            instance.
-	 * @param data
-	 *            data/hash to sign.
-	 * @return signature as hex encoded string.
-	 * @throws NoSuchAlgorithmException
-	 * @throws InvalidKeyException
-	 * @throws SignatureException
-	 */
-	public static String sign(final String algorithm,
-			final PrivateKey privateKey, final byte[] data)
-			throws NoSuchAlgorithmException, InvalidKeyException,
-			SignatureException {
-		Signature signature = Signature.getInstance(algorithm);
-		signature.initSign(privateKey);
-		signature.update(data);
-		return Base64.getEncoder().encodeToString(signature.sign());
-	}
+	/** Signature versions. */
+	public enum Version {
+		SIGNATURE_V1("v1"),
+		SIGNATURE_V2("v2");
 
-	/**
-	 * Verify a signature.
-	 *
-	 * @param algorithm
-	 *            signature algorithm used to generate signature.
-	 * @param publicKey
-	 *            public key corresponding to private key that generated
-	 *            signature.
-	 * @param data
-	 *            the data/hash that was signed.
-	 * @param allegedSignature
-	 *            the signature being verified.
-	 * @return true if computed signature matches allegedSignature.
-	 * @throws NoSuchAlgorithmException
-	 * @throws InvalidKeyException
-	 * @throws SignatureException
-	 * @throws IOException
-	 */
-	public static boolean verify(final String algorithm,
-			final PublicKey publicKey, final byte[] data,
-			final String allegedSignature) throws NoSuchAlgorithmException,
-			InvalidKeyException, SignatureException, IOException {
-		Signature signature = Signature.getInstance(algorithm);
-		signature.initVerify(publicKey);
-		signature.update(data);
-		return signature.verify(Base64.getDecoder().decode(allegedSignature));
+		private String value;
+
+		Version(final String value) {
+			this.value = value;
+		}
+
+		public String toString() {
+			return this.value;
+		}
+
+		public static Version fromString(final String value) throws IllegalArgumentException {
```

Comment:
I don't think this is a checked exception.

Reply:
Switched to javadoc.